cunumeric/_ufunc/ufunc.py

```diff
@@ -557,33 +557,15 @@ def _find_common_type(
         if len(unique_dtypes) == 1 and all_ndarray:
             return arrs[0].dtype
 
-        # FIXME: The following is a miserable attempt to implement type
```

Comment:
@magnatelee you might want to look over this change, as you originally put in this "emulation" code. According to @seberg here https://github.com/nv-legate/cunumeric/pull/972#discussion_r1238344237, this change should be doing the right thing.

Reply:
:+1: I would like some extra eyes on this

Reply:
I'm fine with the change as long as it passes all the tests. and I'd actually go see if we can run this test as well with the change: https://github.com/nv-legate/cunumeric/blob/branch-23.07/tests/integration/test_binary_op_typing.py#L132-L136 I vaguely remember `np.result_type` wasn't doing the right thing in the past and so I ended up with that messy typing code. if that's all sorted out in 1.25 and `np.result_type` is consistent with all the corner cases of numpy's ufuncs, it's great actually and I can't see a reason now to use it.

Reply:
@magnatelee all tests pass locally including those previously `xfail` tests for, except for `test_argsort.py `, butthat also fails for me on the main branch. I'd like to see this pass run in CI  once the build issues are sorted, then I will plan to merge.

Reply:
This change should be backwards and I think also future proof.  The only thing is that it converts the scalars to numpy arrays (and currently has to).

Promotion can differ for a single ufunc sometimes, I added `np.ufunc.resolve_dtypes()` for that, but it opts in to (probable) future behavior so I don't think it's time to seriously consider using it.